Troubleshooting-Guide.md

```diff
@@ -0,0 +1,168 @@
+While the PSAT extension is developed with meticulous attention to detail, the dynamic nature of software and the vast array of user environments mean that unforeseen issues, or "bugs," can emerge. These often arise from unique or previously unencountered scenarios during real-world use.
+
+Software development can be compared to exploring uncharted territory. Though common paths are thoroughly tested, unexpected conditions can lead to the extension behaving in unanticipated ways, potentially causing it to malfunction.
+
+## Reasons for Extension Breakdowns
+
+<img width="1200" alt="PSAT Troubleshooting guide example errors" src="images/troubleshooting-guide/psat_v1.0.0_troubleshooting_guide_error_30_06_2025.png">
+
+Users are essential partners in identifying these elusive bugs. As you use the PSAT extension in your specific development workflows, you may encounter edge cases and scenarios that were not anticipated during its creation. Your real-world usage provides the most effective test for the extension's robustness.
+
+### Idle Chrome Tabs and Service Worker Termination
+
+To ensure smooth performance and prevent excessive memory or CPU consumption, Chrome actively manages system resources. 
+
+This includes identifying and terminating processes in tabs that are not being actively used. Service workers, which are essential for background tasks and offline capabilities, can be resource-intensive if left running indefinitely.
+
+To optimize resource usage, Chrome will terminate service workers that have been idle for a certain period. If a service worker hasn't been actively handling events or tasks, Chrome reclaims the resources it was using.
+
+**How This Impacts the PSAT Extension:**
+
+The PSAT extension relies on its service worker for core functionality (such as data processing or background synchronization), Chrome's termination of an idle service worker can disrupt the extension's operations. 
```

Comment:
```suggestion
The PSAT extension relies on its service worker for core functionality, such as data processing or background synchronization. If Chrome terminates an idle service worker, it can disrupt the extension's operations.
```